File: spring/backend-ms1/src/main/java/ma/zs/stocky/service/facade/admin/achat/AchatAdminService.java
package ma.zs.stocky.service.facade.admin.achat;

import java.util.List;
import ma.zs.stocky.bean.core.achat.Achat;
import ma.zs.stocky.dao.criteria.core.achat.AchatCriteria;
import ma.zs.stocky.zynerator.service.IService;



public interface AchatAdminService extends  IService<Achat,AchatCriteria>  {

    List<Achat> findByClientId(Long id);
    int deleteByClientId(Long id);
    long countByClientCin(String cin);




}

File: spring-angular/roma/v1/frontend/src/app/controller/service/admin/achat/AchatAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {AchatDto} from 'src/app/controller/model/achat/Achat.model';
import {AchatCriteria} from 'src/app/controller/criteria/achat/AchatCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class AchatAdminService extends AbstractService<AchatDto, AchatCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
        this.setApi(environment.apiUrl + 'admin/achat/');
    }

    getModelName(modelName: string): string {
        return this.modeleName = modelName;
    }
    public constrcutDto(): AchatDto {
        return new AchatDto();
    }

    public constrcutCriteria(): AchatCriteria {
        return new AchatCriteria();
    }
}
